### PR TITLE
fantland.cpp : Minor cleanups

### PR DESCRIPTION
**Pull request by cam900**
Reduce ACCESSING_BITs, Runtime tag map lookups, duplicates, Fix some namings, Spacings, remove unnecessary MACHINE_START_CALL_MEMBER, MACHINE_RESET_CALL_MEMBER

### CODE REVIEW

File: src/mame/drivers/fantland.cpp [hunk 1]
```diff
@@ -368,43 +338,38 @@ WRITE8_MEMBER(fantland_state::borntofi_msm5205_w)
...
 }
 
-WRITE_LINE_MEMBER(fantland_state::borntofi_adpcm_int_0) { borntofi_adpcm_int(m_msm1, 0); }
-WRITE_LINE_MEMBER(fantland_state::borntofi_adpcm_int_1) { borntofi_adpcm_int(m_msm2, 1); }
-WRITE_LINE_MEMBER(fantland_state::borntofi_adpcm_int_2) { borntofi_adpcm_int(m_msm3, 2); }
-WRITE_LINE_MEMBER(fantland_state::borntofi_adpcm_int_3) { borntofi_adpcm_int(m_msm4, 3); }
+WRITE_LINE_MEMBER(fantland_state::borntofi_adpcm_int_0) { borntofi_adpcm_int(0); }
+WRITE_LINE_MEMBER(fantland_state::borntofi_adpcm_int_1) { borntofi_adpcm_int(1); }
+WRITE_LINE_MEMBER(fantland_state::borntofi_adpcm_int_2) { borntofi_adpcm_int(2); }
+WRITE_LINE_MEMBER(fantland_state::borntofi_adpcm_int_3) { borntofi_adpcm_int(3); }
```
Comment (cuavas) [inline src/mame/drivers/fantland.cpp]: You could do this with a template rather than the four trampolines.

File: src/mame/includes/fantland.h [hunk 2]
```diff
@@ -53,40 +51,36 @@ class fantland_state : public driver_device
 	required_device<screen_device> m_screen;
 	required_device<palette_device> m_palette;
 	required_device<generic_latch_8_device> m_soundlatch;
-	optional_device<msm5205_device> m_msm1;
-	optional_device<msm5205_device> m_msm2;
-	optional_device<msm5205_device> m_msm3;
-	optional_device<msm5205_device> m_msm4;
+	optional_device_array<msm5205_device, 4> m_msm;
+	optional_region_ptr<uint8_t> m_adpcm_rom;
```
Comment (cuavas) [inline src/mame/includes/fantland.h]: Why not make a derived state class for the systems that have the MSM5205 chips?  Then you can use required device/region finders here and get validation benefits.

**Comment by cuavas** [comment]
This causes a validity error in one set:
```
Driver galaxygn (file fantland.cpp): 1 errors, 0 warnings
Errors:
   Intel I8088 device 'audiocpu': program space memory map entry C0000-FFFFF extends beyond region 'maincpu' size (80000)

Validity check failed (1 errors, 0 warnings in total)
```
Please address the issue.

**Comment by Tafoid** [comment]
To follow up, all machines in the driver now show this upon running:
**Intel I8088 device 'audiocpu': program space memory map entry C0000-FFFFF extends beyond region 'maincpu' size (80000)**
Only galaxygn will show up during -validate.

**Comment by Tafoid** [comment]
My mistake.  galaxygn is the only offender as my program which tests saw the memory region issue text when testing the other sets as well so assumed it was for that machine as well.

**Pull request merged**